Automattic/jetpack
# Emails: Add new menu item for "Inbox", add tests for this operation.

#### Changes proposed in this Pull Request:

This Pull Request is adding a new menu item in the menu for "Inbox", as well as its own test.

#### Does this pull request change what data or activity we track or use?
No, this pull request does not change what data ro activity we track

#### Testing instructions:

For WPCOM simple site:

- Merge the changed files from D67203-code on your sandbox
- Without making any changes on your sandbox, verify that the under Upgrades menu, **_there is NOT_** a new Inbox menu
- Now add the following code to run on your sandbox (0-sandbox.php):
- add_filter( 'jetpack_show_wpcom_inbox_menu', '__return_true' );
- Verify that the navigation menu under Upgrades contains now a Inbox menu

For wPCOM atomic site:

- Ensure that you have the [Jetpack Beta](https://jetpack.com/download-jetpack-beta/) plugin uploaded and installed on your Atomic test site, as well as a mechanism for running code on the site. (A plugin like [Code Snippets](https://wordpress.org/plugins/code-snippets/) can make this relatively easy.)
- Use Jetpack Beta to switch to this branch: add/new-menu-item-for-inbox-functionality
- Refresh the admin page you're on, and verify that below Upgrade menu there is no Inbox menu
- Ensure that the following code is run on your site:
- add_filter( 'jetpack_show_wpcom_inbox_menu', '__return_true' );
- Refresh the admin page you're on, and verify that below Upgrade menu there is an Inbox menu

**Screenshots**
![image](https://user-images.githubusercontent.com/5689927/134323002-054ee80a-ead2-42c3-b734-8541b0635768.png)
![image](https://user-images.githubusercontent.com/5689927/134382720-caaf4f67-bb57-4c5a-9de6-9b991b43d4b1.png)



**Remarks**
To test that it works correctly some patches in calypso and wpcom must be applied (or await until they are deployed).
wpcom: Patch D66920
calypso: https://github.com/Automattic/wp-calypso/pull/56429

Related to {1200182182542585-as-1200979289212227} and {1200182182542585-as-1200979289212229}
